File: FeedItForward/backend/app/schemas/pickup_job.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum

from .leftover_food import LeftoverFood
from .consumer import Consumer
from .driver import Driver
from .misc import Geometry

class PickupJobStatus(Enum):
    AVAILABLE = "Available"
    IN_PROGRESS = "In Progress"
    COMPLETED = "Completed"

class PickupJob(BaseModel):
    pickup_job_id: int

    start_location: Geometry
    end_location: Geometry
    description: str
    status: PickupJobStatus
    photo_proofs: list[str]

    leftover_food_id: int
    leftover_food: LeftoverFood

    consumer_id: int
    consumer: Consumer

    driver_id: Optional[int] = None
    driver: Optional[Driver] = None

    class Config:
        orm_mode = True

class PickupJobCreate(BaseModel):
    start_location: Geometry
    end_location: Geometry
    description: str
    status: PickupJobStatus

    leftover_food_id: int
    consumer_id: int

class PickupJobUpdate(BaseModel):
    pickup_job_id: int
    start_location: Geometry
    end_location: Geometry
    description: str
    status: PickupJobStatus
    photo_proofs: list[str]

    driver_id: int

File: FeedItForward/frontend/src/schemas/pickupJob.ts
import { Consumer } from "./consumer";
import { Driver } from "./driver";
import { LeftoverFood } from "./leftoverFood";
import { Geometry } from "./misc";

export const PICKUP_JOB_ACTION = ["Accept", "Ignore", "Complete"] as const;
export type PickupJobAction = (typeof PICKUP_JOB_ACTION)[number];

export const PICKUP_JOB_STATUS = [
  "Available",
  "In Progress",
  "Completed"
] as const;
export type PickupJobStatus = (typeof PICKUP_JOB_STATUS)[number];

export interface PickupJob {
  pickup_job_id: number;

  start_location: Geometry;
  end_location: Geometry;
  description: string;
  status: PickupJobStatus;
  photo_proofs: string[];

  leftover_food_id: number;
  leftover_food: LeftoverFood;

  consumer_id: number;
  consumer: Consumer;

  driver_id?: number;
  driver?: Driver;
}

export interface PickupJobCreate {
  start_location: Geometry;
  end_location: Geometry;
  description: string;
  status: PickupJobStatus;

  leftover_food_id: number;
  consumer_id: number;
}

export interface PickupJobUpdate {
  pickup_job_id: number;

  start_location: Geometry;
  end_location: Geometry;
  description: string;
  status: PickupJobStatus;
  photo_proofs: string[];

  driver_id: number;
}
